zmon-controller-app/src/main/java/org/zalando/zmon/controller/MultiKairosDBController.java

```diff
@@ -89,15 +115,59 @@ public MultiKairosDBController(KairosDBProperties kairosDBProperties, MetricRegi
             q.put("cache_time", 60);
             if (q.has("start_absolute")) {
                 long start = q.get("start_absolute").asLong();
+                long finish = q.get("end_absolute").asLong();
+                queryDistance = (System.currentTimeMillis() - start)/86400000;
+                querySpan = (finish-start)/86400000;
                 start = start - (start % 60000);
                 q.put("start_absolute", start);
             }
             else if (q.has("start_relative")) {
+
+                ObjectNode r = (ObjectNode)q.get("start_relative");
+
                 if(queryWindow != 0) {
-                    ObjectNode r = (ObjectNode)q.get("start_relative");
                     r.put("value", queryWindow);
                     r.put("unit", "minutes");
                 }
+
+                switch(r.get("unit").asText()){
+                    case "seconds":
+                        querySpan = r.get("value").asLong()/86400;
+                        break;
+                    case "minutes":
+                        querySpan = r.get("value").asLong()/1440;
+                        break;
+                    case "hours":
+                        querySpan = r.get("value").asLong()/24;
+                        break;
+                    case "days":
+                        querySpan = r.get("value").asLong();
+                        break;
+                    case "weeks":
+                        querySpan = r.get("value").asLong()*7;
+                        break;
+                    case "months":
+                        querySpan = r.get("value").asLong()*30;
+                        break;
+                    case "years":
+                        querySpan = r.get("value").asLong()*365;
+                        break;
+                    default:
+                        throw new IllegalArgumentException("Invalid unit: " + r.get("unit").toString());
+                }
+            }
+
+            if (querySpan >= config.queryDays) {
+                span.setTag("LongQuery", "True");
+                span.setTag("Check Id", checkId);
+                span.setTag("Referer", referer);
+                span.setTag("Query-Span", querySpan + " days");
+            }
+            if (queryDistance >= config.queryDays) {
+                span.setTag("LongQuery", "True");
+                span.setTag("Check Id", checkId);
+                span.setTag("Referer", referer);
+                span.setTag("Query-Distance", queryDistance + " days");
             }
```

Comment:
In regards to the tags names, I'm also a fan of lower case tags, following the conventions from the [standard OpenTracing span tags](https://github.com/opentracing/specification/blob/master/semantic_conventions.md#standard-span-tags-and-log-fields).